packages/xo-server-perf-alert/src/xapiPerfmon.js

```diff
@@ -0,0 +1,214 @@
+// @ts-check
+
+import { asyncEach } from '@vates/async-each'
+import { Alarm, MonitorStrategy } from './Strategy.js'
+
+/**
+ * @import {XoMessage, XoHost, XoSr, XoVm} from "@vates/types"
+ * @import { MonitorRuleSet } from "./Rules.js"
+ *
+ */
+
+const OTHER_PROPERTY_NAME = {
+  host: 'otherConfig',
+  SR: 'other_config',
+  VM: 'other',
+}
+const TO_PERFMON_TYPE = {
+  cpuUsage: 'cpu_usage',
+  memoryUsage: 'mem_usage',
+  storageUsage: 'fs_usage',
+}
+const TO_VARIABLE_NAME = {
+  cpu_usage: 'cpuUsage',
+  mem_usage: 'memoryUsage',
+  fs_usage: 'storageUsage',
+}
+
+const TO_SENSE = {
+  '<': 'low',
+  '>': 'high',
+}
+export class XapiPerfmon extends MonitorStrategy {
+  #xo
+  /**
+   * @type {MonitorRuleSet}
+   */
+  #rules
+
+  /**
+   * @type {NodeJS.Timeout|undefined}
+   */
+  #watchInterval
+
+  /**
+   *
+   * @param {*} xo
+   * @param {MonitorRuleSet} rules
+   */
+  constructor(xo, rules) {
+    super()
+    this.#xo = xo
+    this.#rules = rules
+    this.updateObjectsPerfmon = this.updateObjectsPerfmon.bind(this)
+  }
+
+  /**
+   *
+   * @returns {Promise<Array<Alarm>>}
+   */
+  async computeActiveAlarms() {
+    const alarms = []
+    /**
+     * @type {Array<XoMessage>}
+     */
+    const xapiAlarms = Object.values(this.#xo.objects.indexes.type.message ?? [])
+    for (const xapiAlarm of xapiAlarms) {
```

Comment:
In my opinion, if you need to retrieve alarms, you could move `alarm.service.mts#getAlarms`.

This will simplify the code and ensure that we always handle alarms in the same way.